### ISSUE TITLE
Issue opening Game Maker Project

### ISSUE THREAD

**Issue by briviere** [IsmAvatar/LateralGM#324]
Hello 

I was trying to import the attached Game Maker Project.  LateralGM is throwing the following exception:

[Blaster Faster.zip](https://github.com/IsmAvatar/LateralGM/files/691520/Blaster.Faster.zip)

Operating System: Linux
Version: 3.13.0-37-generic
Architecture: amd64

Java Name: Java HotSpot(TM) 64-Bit Server VM
Java Vendor: Oracle Corporation
Version: 1.8.0_111

Available processors (cores): 2
Free memory (bytes): 67243184
Maximum memory (bytes): 1370488832
Total memory available to JVM (bytes): 119013376

File system root: /
Total space (bytes): 487695884288
Free space (bytes): 236194115584
Usable space (bytes): 211396972544

Stack trace:
org.lateralgm.file.GmFormatException: java.io.IOException: Unexpected end of file reached at filepos: 0
	at org.lateralgm.file.StreamDecoder.read(StreamDecoder.java:76)
	at org.lateralgm.file.StreamDecoder.read2(StreamDecoder.java:84)
	at org.lateralgm.file.iconio.ICOFile.readHeader(ICOFile.java:167)
	at org.lateralgm.file.iconio.ICOFile.read(ICOFile.java:156)
	at org.lateralgm.file.iconio.ICOFile.<init>(ICOFile.java:123)
	at org.lateralgm.file.iconio.ICOFile.<init>(ICOFile.java:108)
	at org.lateralgm.file.GMXFileReader.readConfigurations(GMXFileReader.java:446)
	at org.lateralgm.file.GMXFileReader.readProjectFile(GMXFileReader.java:274)
	at org.lateralgm.file.GMXFileReader.readProjectFile(GMXFileReader.java:200)
	at org.lateralgm.main.FileChooser$ProjectReader.read(FileChooser.java:385)
	at org.lateralgm.main.FileChooser$1.run(FileChooser.java:588)
	at java.lang.Thread.run(Thread.java:745)

java.lang.NullPointerException
	at javax.swing.ImageIcon.<init>(ImageIcon.java:240)
	at org.lateralgm.subframes.GameSettingFrame.setIconPreviewToGameIcon(GameSettingFrame.java:636)
	at org.lateralgm.subframes.GameSettingFrame.setComponents(GameSettingFrame.java:1005)
	at org.lateralgm.subframes.GameSettingFrame.revertResource(GameSettingFrame.java:1032)
	at org.lateralgm.main.LGM.reload(LGM.java:915)
	at org.lateralgm.main.FileChooser.open(FileChooser.java:621)
	at org.lateralgm.main.FileChooser.open(FileChooser.java:568)
	at org.lateralgm.main.FileChooser.openNewFile(FileChooser.java:525)
	at org.lateralgm.main.FileChooser.open(FileChooser.java:557)
	at org.lateralgm.main.Listener.actionPerformed(Listener.java:309)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.AbstractButton.doClick(AbstractButton.java:376)
	at javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:833)
	at javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:877)
	at java.awt.Component.processMouseEvent(Component.java:6533)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6298)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4525)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
	at java.awt.Container.dispatchEventImpl(Container.java:2280)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

**Comment by Hughtech** [comment]
This maybe nothing but maybe you should update JAVA to version 8. I installed Ubuntu on 2 laptops and tried using Java 6 and 7, which resulted in errors when I tried to run ENIGMA. I also installed WINE and PLAYONLINUX and installed the Windows version. Still no luck. I tried ENIGMA on my MX LINUX Desktop system and it worked perfectly. I found out that JAVA 8 was installed as part of the system.
I'm just saying.
Hope this helped.

**Comment by RobertBColton** [comment]
Ok I see a couple of issues here, but otherwise your Java version is fine. You are actually using an old version of LGM, because I get a much different error log:
```
Stack trace:
org.lateralgm.file.GmFormatException: failed to read: C:\Users\Owner\Desktop\Blaster.Faster/Configs\Default\windows\runner_icon.ico
	at org.lateralgm.file.GMXFileReader.readConfigurations(GMXFileReader.java:470)
	at org.lateralgm.file.GMXFileReader.readProjectFile(GMXFileReader.java:295)
	at org.lateralgm.file.GMXFileReader.readProjectFile(GMXFileReader.java:233)
	at org.lateralgm.main.FileChooser$ProjectReader.read(FileChooser.java:385)
	at org.lateralgm.main.FileChooser$1.run(FileChooser.java:588)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.io.FileNotFoundException: C:\Users\Owner\Desktop\Blaster.Faster\Configs\Default\windows\runner_icon.ico (The system cannot find the path specified)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(Unknown Source)
	at java.base/java.io.FileInputStream.<init>(Unknown Source)
	at java.base/java.io.FileInputStream.<init>(Unknown Source)
	at org.lateralgm.file.StreamDecoder.<init>(StreamDecoder.java:45)
	at org.lateralgm.file.iconio.ICOFile.<init>(ICOFile.java:75)
	at org.lateralgm.file.GMXFileReader.readConfigurations(GMXFileReader.java:466)
	... 5 more
```

Basically, the icon file for your game is actually missing. The old GMX reader I wrote isn't so helpful in letting you know about it though. I just copied the missing folder over from another GMX project and your project loads fine.

I am going to make some changes to address this.

**Comment by RobertBColton** [comment]
Closing as resolved by #332, using the default is the best LGM can do in this situation to handle the game icon being missing. You can grab the 1.8.10 release with this fix on the releases page: https://github.com/IsmAvatar/LateralGM/releases
